src/libraries/System.Text.Json/tests/Serialization/CustomConverterTests/CustomConverter.Dynamic.Sample.cs

```diff
@@ -0,0 +1,444 @@
+    /// </remarks>
+    public static class JsonSerializerExtensions
+    {
+        /// <summary>
+        /// Enable support for the <see langword="dynamic"/> feature.
+        /// Changes the default handling for types specified as <see cref="object"/> from deserializing as
+        /// <see cref="System.Text.Json.JsonElement"/> to instead deserializing as the one of the
+        /// <see cref="JsonDynamicType"/>-derived types including:
+        /// <see cref="JsonDynamicObject"/>,
+        /// <see cref="JsonDynamicArray"/>,
+        /// <see cref="JsonDynamicString"/>,
+        /// <see cref="JsonDynamicNumber"/> and
+        /// <see cref="JsonDynamicBoolean"/>.
+        /// </summary>
+        /// <remarks>
+        /// When deserializing <see cref="System.Text.Json.JsonTokenType.StartObject"/>, <see cref="JsonDynamicObject"/>
+        /// is returned which implements <see cref="System.Collections.IDictionary{string, object}"/>.
+        /// When deserializing <see cref="System.Text.Json.JsonTokenType.StartArray"/>, <see cref="System.Collections.IList{object}"/>
+        /// is returned which implements <see cref="System.Collections.IList{object}"/>.
+        /// When deserializing <see cref="System.Text.Json.JsonTokenType.String"/>, <see cref="JsonDynamicString"/>
+        /// is returned and supports an implicit cast to <see cref="string"/>.
+        /// An explicit cast or assignment to other types, such as <see cref="System.Text.Json.JsonTokenType.DateTime"/>,
+        /// is supported provided there is a custom converter for that Type.
+        /// When deserializing <see cref="System.Text.Json.JsonTokenType.Number"/>, <see cref="JsonDynamicNumber"/> is returned.
+        /// An explicit cast or assignment is required to the appropriate number type, such as <see cref="decimal"/> or <see cref="long"/>.
+        /// When deserializing <see cref="System.Text.Json.JsonTokenType.True"/> and <see cref="System.Text.Json.JsonTokenType.False"/>,
+        /// <see cref="JsonDynamicBool"/> is returned and supports an implicit cast to <see cref="bool"/>.
+        /// An explicit cast or assignment to other types is supported provided there is a custom converter for that type.
+        /// When deserializing <see cref="System.Text.Json.JsonTokenType.Null"/>, <see langword="null"/> is returned.
+        /// </remarks>
+        public static void EnableDynamicTypes(this JsonSerializerOptions options)
+        {
+            options.Converters.Add(new DynamicObjectConverter());
+        }
+
+        /// <summary>
+        /// The base class for all dynamic types supported by the serializer.
+        /// </summary>
+        public abstract class JsonDynamicType : DynamicObject
+        {
+            public JsonSerializerOptions Options { get; private set; }
+
+            internal JsonDynamicType(JsonSerializerOptions options)
+            {
+                Options = options;
```

Comment:
As an extension, no, since the default value can't be obtained.

If built-in to STJ, perhaps, since we can obtain the default.

I'll add the check, however, I won't be adding full XML doc until this is productized.